### PR TITLE
Create: 0093-restore-ip-addresses

### PR DESCRIPTION
**Pull request by AkifhanIlgaz**
[//]: # "Pull Request Template"
[//]: # "Replace the placeholder values in the template below"

- **File(s) Modified**: _0093-restore-ip-addresses.rs, 0093-restore-ip-addresses.ts, 0093-restore-ip-addresses.js, 0093-restore-ip-addresses.go_
- **Language(s) Used**: _Rust, Typescript, Javascript, Go_
- **Submission URL**: 
-_Rust: https://leetcode.com/problems/restore-ip-addresses/submissions/882653196/_
-_Typescript: https://leetcode.com/problems/restore-ip-addresses/submissions/882630377/_
-_Javascript: https://leetcode.com/problems/restore-ip-addresses/submissions/882630668/_
-_Go: https://leetcode.com/problems/restore-ip-addresses/submissions/882625107/_ 

[//]: # "Getting the Submission URL"
[//]: # "Go to the leetcode [`Submissions tab`](https://user-images.githubusercontent.com/71089234/180188604-b1ecaf90-bf27-4fd6-a559-5567aebf8930.png)"
[//]: # "and [click on the `Accepted` status of your submission.](https://user-images.githubusercontent.com/71089234/180189321-1a48c33f-aa65-4b29-8aaa-685f4f5f8c9e.png)]"
[//]: # "Finally copy the URL from the nav bar, it should look like https://leetcode.com/problems/[problem-name]/submissions/xxxxxxxxx/"

### Important
Please make sure the file name is lowercase and a duplicate file does not already exist before merging.

### CODE REVIEW

File: javascript/0093-restore-ip-addresses.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                backtracking(
+                    j + 1,
+                    dots + 1,
+                    currentIP + s.slice(i, j + 1) + '.'
+                );
+            }
+        }
+    }
+
+    backtracking(0, 0, '');
```
Comment (aakhtar3) [inline javascript/0093-restore-ip-addresses.js:35]: I posted this in another submission and it applies here too. Btw thanks for doing multiple solutions!!

https://github.com/neetcode-gh/leetcode/pull/2114#discussion_r1083339676

Comment (AkifhanIlgaz) [reply javascript/0093-restore-ip-addresses.js:35]: I tried to implement the same solution in the video

Comment (aakhtar3) [reply javascript/0093-restore-ip-addresses.js:35]: Can you post link to video?

Comment (aakhtar3) [reply javascript/0093-restore-ip-addresses.js:35]: Ahh ok just watched the video! This problem can be solved in many ways. Backtrack is a special type of DFS where you try a "path" then "undo" and try a new "path" recursively.

Take a peek at the neatcode 150 python solutions for backtrack. Most follow that particular pattern that I mentioned.

Thoughts @neetcode-gh ?

**Review by tahsintunan — APPROVED** [review]

**Pull request merged**